src/views/FilesSidebarChatView.vue

```diff
@@ -23,7 +23,7 @@
 	<div class="talk-tab__wrapper">
 		<CallButton class="call-button" />
 		<ChatView />
-		<MediaSettings :initialize-on-mounted="false" />
+		<MediaSettings :initialize-on-mounted="false" :recording-consent-given.sync="recordingConsentGiven" />
```

Comment:
So why are you adding `recordingConsentGiven` here ?

Reply:
to be able to connect from files sidebar also, if consent is set to 1 (applied to all calls on instance)
consent may be reused when reconnecting, so should not be stored in MediaSettings component. Also don't want to get a new store for it

Reply:
For "configurable" mode, consent will be switched off for new conversations like this, so not a problem
For "required" mode, I'd say that it's a pre-emptive measure (to not come back to it when we add owners/moderators to such conversations)

Reply:
To keep this pre-emptive measure, consent should not be mandatory atm as this might be a GDPR violation https://gdpr.eu/gdpr-consent-requirements/ (see "Consent must be freely given" section)  
> According to [Recital 42](https://gdpr.eu/Recital-42-Burden-of-proof-and-requirements-for-consent/), “Consent should not be regarded as freely given if the data subject has no genuine or free choice or is unable to refuse or withdraw consent without detriment.” 
The one exception is if you need some piece of data from someone to provide them with your service. For example, you may need their credit card information to process a transaction or their mailing address to ship a product.


For those participants who do not wish to consent yet, they want to join the call, you are mandating their consent despite that it does not fall under the one exception because you don't need their consent in this scenario.

Suggested alternative : An opt-in choice to consent or decline without impacting their ability to join the call

Reply:
> For those participants who do not wish to consent yet, they want to join the call, you are mandating their consent despite that it does not fall under the one exception because you don't need their consent in this scenario.

This is a customer requirement atm. It is not possible to join without giving consent on API level and this was approved by the customer.

> Suggested alternative : An opt-in choice to consent or decline without impacting their ability to join the call

This can be looked at as a potential follow up, but has quite a long trail (settings needs to be stored on a session based level, recording backend needs access to the user list and filter out peoples audio/video where consent was off, etc). Far out of scope for now, so this is what it will be `¯\_(ツ)_/¯` It's not getting worse then before where the recording was always done without even asking.

Reply:
> For those participants who do not wish to consent yet, they want to join the call, you are mandating their consent despite that it does not fall under the one exception because you don't need their consent in this scenario.

That being said, the user can still disable the microphone/camera and will not be recorded. So could say that this is the users choice here and we are basically good again.

Reply:
It's clear that you're talking about file sharing conversations.
Although there are not any owners / moderators to be able to start the recording, technically it's possible (and we won't return to this code again)
Besides, it's possible to promote user with `occ talk:room:promote [token] [use]`

Reply:
Thanks for sharing that @Antreesy . In that case, can end this legal concerns discussion 🧑🏻‍⚖️for now.